File: python/1921. Eliminate Maximum Number of Monsters.py

from ast import List


class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        n = len(dist)
        arrivalTimes = [dist[i]/speed[i] for i in range(n)]
        arrivalTimes = sorted(arrivalTimes)
        index = 0
        while index < n and arrivalTimes[index] - index > 0:
            index += 1
        return min(index , n)

File: javascript/1921. Eliminate Maximum Number of Monsters.js
/**
 * @param {number[]} dist
 * @param {number[]} speed
 * @return {number}
 */
var eliminateMaximum = function (dist, speed) {
  const n = dist.length;
  var arrivalTime = new Array(n);
  for (var i = 0; i < n; i++) {
    arrivalTime[i] = dist[i] / speed[i];
  }
  arrivalTime.sort((a, b) => a - b);
  var ans = 0;
  while (ans < n && arrivalTime[ans] - ans > 0) {
    ans += 1;
  }
  return ans;
};
